File: ncia/day_02_03_scroll/day_02_03_scroll/ScaleView.m
//
//  ScaleView.m
//  day_02_03_scroll
//
//  Created by 최철동 on 2016. 8. 18..
//  Copyright © 2016년 Burke Choi. All rights reserved.
//

#import "ScaleView.h"

@implementation ScaleView

- (instancetype) initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:frame]) {
        UIScrollView* scroll = [[UIScrollView alloc] initWithFrame:self.bounds];
        
        
        scroll.delegate = self;
        scroll.minimumZoomScale = 0.1;
        scroll.maximumZoomScale = 10;
        
        [self addSubview:scroll];
        
        
        mImg = [[UIImageView alloc] initWithFrame:self.bounds];
        mImg.contentMode = UIViewContentModeScaleAspectFit; // 원본 형태로
//        mImg.contentMode = UIViewContentModeScaleToFill; // default option
//        mImg.contentMode = UIViewContentModeCenter; // center 는 확대 축소 되지 않고 가운데에 배치 한다.
     
        [scroll addSubview: mImg];
        
        mLabel = [[UILabel alloc] initWithFrame:self.bounds];
        mLabel.text = @"iPhone";
        mLabel.textAlignment = NSTextAlignmentCenter;
        
        [mImg addSubview:mLabel];
    }
    
    
    
    return self;
}

- (void) setImage: (UIImage*) img {
    mImg.image = img;
}

- (nullable UIView *)viewForZoomingInScrollView:(UIScrollView *)scrollView {
    return mImg;
}
@end

File: ncia/day02_05_photoSlider/ScaleView.swift
//
//  ScaleView.swift
//  day_02_04_scrollview_swift
//
//  Created by 최철동 on 2016. 8. 18..
//  Copyright © 2016년 Burke Choi. All rights reserved.
//

import UIKit

class ScaleView: UIView, UIScrollViewDelegate {
    
    // let 은 상수 , var 은 일반 변수
    var mIv: UIImageView? = nil
    
    override init(frame: CGRect) {
        super.init(frame: frame);
        
        let scroll = UIScrollView(frame: self.bounds)
        scroll.delegate = self
        scroll.minimumZoomScale = 0.1
        scroll.maximumZoomScale = 10
        
        self.addSubview(scroll)
        
        mIv = UIImageView(frame: self.bounds)
//        mIv?.contentMode = UIViewContentMode.UIViewContentModeScaleAspectFit
        mIv?.contentMode = UIViewContentMode.ScaleAspectFit
        scroll.addSubview(mIv!)
        
//        self.addSubview(mIv!) // ! 는 force (강제성 nil 일 수도 있기 때문에)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init (coder:) has not been implemeted")
    }
    
    func setImage(image: UIImage) {
        mIv?.image = image;
    }
    
    func viewForZoomingInScrollView(scrollView: UIScrollView) -> UIView? {
        
        return mIv;
    }
}
